firmware/applications/SignBuddy/comms.c

```diff
@@ -0,0 +1,174 @@
+#include "comms.h"
+
+#include "ble_uart.h"
+#include "board.h"
+#include "common.h"
+#include "crc.h"
+#include "fifo.h"
+#include "logger.h"
+#include "pb_encode.h"
+#include "SignBuddy.pb.h"
+
+#include <stdlib.h>
+
+#define COMMS_TX_BUFFER_SIZE    512
+
+#define SYNC_BEGIN              0x16
+#define SYNC_END                0x27
+
+#define MID_SAMPLE              0x01
+
+#define PACKET_SIZE             sizeof(packet_header_t) + s.packet_header.length * sizeof(uint8_t) + \
+  sizeof(packet_footer_t)
+
+typedef struct __attribute__((__packed__)) {
+  uint8_t sync_begin;
+  uint8_t mid;
+  uint8_t length;
+} packet_header_t;
+
+typedef struct __attribute__((__packed__)) {
```

Comment:
I liked the way you had it before with the packet struct

    typedef struct {
      uint8_t sync;
      uint8_t length;
      uint8_t crc;
      uint8_t data[MAX_SIZE]
     }

Reply:
was there a particular reason you changed it?

Reply:
Switched back, [170f39b](https://github.com/adambujak/SignBuddy/pull/97/commits/170f39b8a02767f49f0a524c787bdb71232e7bd5)